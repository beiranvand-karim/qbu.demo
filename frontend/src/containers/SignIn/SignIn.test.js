import {shallow} from 'enzyme'
import React from "react"
import {SignIn} from "./"
import renderer from 'react-test-renderer'
import {MemoryRouter} from "react-router-dom"

describe('<SignIn/>', () => {

   const checkSignInBeginMock = jest.fn();
   const checkSignInSuccessMock = jest.fn();
   const checkSignInErrorMock = jest.fn();

   it('should render without crashing', () => {
      shallow(<SignIn/>)
   });

   it('should sign in', () => {
      window.fetch = jest.fn().mockImplementation(() => {
            return new Promise((resolve) => {
               resolve({
                  status: 200,
                  json: () => {
                     return new Promise((resolve) => {
                        resolve({
                           token: 'cc0f80d7-ca5c-4c97-aa45-8ee02a5b80de'
                        })
                     })
                  }
               })
            })
         }
      );

      const wrapper = shallow(<SignIn
         checkSignInBegin={checkSignInBeginMock}
         checkSignInSuccess={checkSignInSuccessMock}
         checkSignInError={checkSignInErrorMock}/>);

      wrapper.find('Form').simulate('submit', {preventDefault() {}});

      expect(checkSignInBeginMock.mock.calls.length).toEqual(1);
      setImmediate(()=>{
         expect(checkSignInSuccessMock.mock.calls.length).toEqual(1);
      })

   });

   it('should call on sign in fail', function () {
      window.fetch = jest.fn().mockImplementation(() => {
            return new Promise((reject) => {
               reject({
                  error: 'sign in failed.'
               })
            })
         }
      );

      const wrapper = shallow(<SignIn
         checkSignInBegin={checkSignInBeginMock}
         checkSignInSuccess={checkSignInSuccessMock}
         checkSignInError={checkSignInErrorMock}/>);

      wrapper.find('Form').simulate('submit', {preventDefault() {}});

      expect(checkSignInBeginMock.mock.calls.length).toEqual(2);
      setImmediate(()=>{
         expect(checkSignInErrorMock.mock.calls.length).toEqual(1);
      })

   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><SignIn/></MemoryRouter>).toJSON();
      expect(tree).toMatchSnapshot()
   });

});
