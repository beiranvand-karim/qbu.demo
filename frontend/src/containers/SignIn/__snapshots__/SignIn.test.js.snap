// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn/> should matches the snapshot 1`] = `
<div>
  <div
    className="SignInFormContainer"
  >
    <form
      className="SignInForm"
      onSubmit={[Function]}
    >
      <div
        className="formGroup position-relative form-group"
      >
        <label
          className=""
          htmlFor="userName"
        >
          user name
        </label>
        <input
          className="formGroup formGroup_input form-control"
          id="userName"
          name="userName"
          onChange={[Function]}
          placeholder="user name"
          type="text"
          value=""
        />
      </div>
      <div
        className="formGroup position-relative form-group"
      >
        <label
          className=""
          htmlFor="passWord"
        >
          pass word
        </label>
        <input
          className="formGroup formGroup_input form-control"
          id="passWord"
          name="passWord"
          onChange={[Function]}
          placeholder="pass word"
          type="text"
          value=""
        />
      </div>
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
